Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp

```diff
@@ -539,6 +541,81 @@ static void addSubresourcesForAttachmentElementsIfNecessary(LocalFrame& frame, c
 
 #endif
 
+static HashMap<RefPtr<CSSStyleSheet>, String> addSubresourcesForCSSStyleSheetsIfNecessary(LocalFrame& frame, const String& subresourcesDirectoryName, HashSet<String>& uniqueFileNames, HashMap<String, String>& uniqueSubresources, Vector<Ref<ArchiveResource>>& subresources)
+{
+    if (subresourcesDirectoryName.isEmpty())
+        return { };
+
+    auto document = frame.protectedDocument();
+    if (!document)
+        return { };
+
+    HashMap<RefPtr<CSSStyleSheet>, String> uniqueCSSStyleSheets;
+    HashMap<RefPtr<CSSStyleSheet>, String> relativeUniqueCSSStyleSheets;
+    auto& documentStyleSheets = document->styleSheets();
+    for (unsigned index = 0; index < documentStyleSheets.length(); ++index) {
+        RefPtr styleSheet = documentStyleSheets.item(index);
+        if (!styleSheet || !is<CSSStyleSheet>(styleSheet.get()))
+            continue;
+
+        CSSStyleSheet& cssStyleSheet = downcast<CSSStyleSheet>(*styleSheet);
+        if (uniqueCSSStyleSheets.contains(&cssStyleSheet))
+            continue;
+
+        HashSet<RefPtr<CSSStyleSheet>> cssStyleSheets;
+        cssStyleSheets.add(&cssStyleSheet);
+        cssStyleSheet.getChildStyleSheets(cssStyleSheets);
+        for (auto& currentCSSStyleSheet : cssStyleSheets) {
+            bool isExternalStyleSheet = !currentCSSStyleSheet->href().isEmpty() || currentCSSStyleSheet->ownerRule();
+            if (!isExternalStyleSheet)
+                continue;
+
+            auto url = currentCSSStyleSheet->baseURL();
+            if (url.isNull() || url.isEmpty())
+                continue;
+
+            auto addResult = uniqueCSSStyleSheets.add(currentCSSStyleSheet, emptyString());
+            if (!addResult.isNewEntry)
+                continue;
+
+            // Delete cached resource for this style sheet.
+            auto index = subresources.findIf([&](auto& subresource) {
+                return subresource->url() == url;
+            });
+            if (index != notFound) {
+                auto fileName = FileSystem::lastComponentOfPathIgnoringTrailingSlash(subresources[index]->relativeFilePath());
+                uniqueFileNames.remove(fileName);
+                uniqueSubresources.remove(url.string());
+                subresources.remove(index);
+            }
+
+            String subresourceFileName = generateValidFileName(url, uniqueFileNames);
+            uniqueFileNames.add(subresourceFileName);
+            String subresourceFilePath = FileSystem::pathByAppendingComponent(subresourcesDirectoryName, subresourceFileName);
+            addResult.iterator->value = frame.isMainFrame() ? subresourceFilePath : subresourceFileName;
```

Comment:
`addResult.iterator` is an iterator inside `uniqueCSSStyleSheets`. At line 588, you modified `uniqueCSSStyleSheets` so you cannot rely on your iterator still being valid here.